src/crafting.cpp

```diff
@@ -1746,219 +1791,277 @@ void set_item_inventory(item &newit)
     }
 }
 
+/* selection of component if a recipe requirement has multiple options (e.g. 'duct tap' or 'welder') */
+comp_selection player::select_component(const std::vector<item_comp> &components, int batch, bool can_cancel) {
+	std::vector<item_comp> player_has;
+	std::vector<item_comp> map_has;
+	std::vector<item_comp> mixed;
+
+	comp_selection selected = comp_selection();
+	inventory map_inv;
+	map_inv.form_from_map(pos3(), PICKUP_RANGE);
```

Comment:
I think I understand what you mean, but it's really not my territory. When I started making changes this creation of ```map_inv``` was already being done for every component. It just looks weird in the diff because I encapsulated the selecting of components in a separate function.